cranelift/filetests/filetests/isa/x64/b1.clif

```diff
@@ -73,3 +73,134 @@ block2:
 ;   popq    %rbp
 ;   ret
 
+function %test_x_slt_0_i64(i64) -> b1 {
+block0(v0: i64):
+    v1 = iconst.i64 0
+    v2 = icmp slt v0, v1
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarq    $63, %rdi, %rdi
+;   movq    %rdi, %rax
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %test_x_slt_0_i32f4(i32) -> b1 {
+block0(v0: i32):
+    v1 = iconst.i32 0
+    v2 = icmp slt v0, v1
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarl    $31, %edi, %edi
+;   movq    %rdi, %rax
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %test_0_sgt_x_i64(i64) -> b1 {
+block0(v0: i64):
+    v1 = iconst.i64 0
+    v2 = icmp sgt v1, v0
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarq    $63, %rdi, %rdi
+;   movq    %rdi, %rax
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %test_0_sgt_x_i32f4(i32) -> b1 {
+block0(v0: i32):
+    v1 = iconst.i32 0
+    v2 = icmp sgt v1, v0
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarl    $31, %edi, %edi
+;   movq    %rdi, %rax
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %test_0_sle_x_i64(i64) -> b1 {
+block0(v0: i64):
+    v1 = iconst.i64 0
+    v2 = icmp sle v1, v0
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarq    $63, %rdi, %rdi
+;   notq    %rdi, %rdi
+;   movq    %rdi, %rax
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %test_0_sle_x_i32f4(i32) -> b1 {
+block0(v0: i32):
+    v1 = iconst.i32 0
+    v2 = icmp sle v1, v0
+    return v2
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   sarl    $31, %edi, %edi
+;   notl    %edi, %edi
```

Comment:
Is it correct? I guess it should be:
```asm
notl    %edi, %edi
shrl    $31, %edi, %edi
```
https://godbolt.org/z/s1fccc5Ex

Reply:
I've inverted it and restricted this optimization to boolean-producing instructions. Also this shouldn't affect the `test/jmp` fusion, as this optimization will only fire for `icmp` instructions whose results are used directly, not as the input to a branch decision. Was there another fusion case you were thinking this might invalidate?

Reply:
I meant macro / micro fusion [on the CPU side](https://easyperf.net/blog/2018/02/23/MacroFusion-in-Intel-CPUs)